salvo/src/lib/benchmark/base_benchmark.py

```diff
@@ -78,20 +91,54 @@ def run_image(self, image_name : str, **kwargs) -> bytearray:
           here https://docker-py.readthedocs.io/en/stable/containers.html#docker.models.containers.ContainerCollection.run
 
     Returns:
-        A bytearray containing the output produced from executing the specified container
+        A bytearray containing the output produced from executing the specified
+        container
     """
-    pass
+    return self._docker_image.run_image(image_name, **kwargs)
 
   def pull_images(self) -> List[str]:
     """Retrieve all images necessary for the benchmark.
 
-    Retrieve the NightHawk and Envoy images defined in the control object. 
+    Retrieve the NightHawk and Envoy images defined in the control
```

Comment:
Can we add a "Returns:" section here explaining the return value? It also looks like there is a specific case when we return an empty list. Can we document that too in the "Returns" section?

https://github.com/google/styleguide/blob/gh-pages/pyguide.md#383-functions-and-methods

Reply:
Updated

Reply:
Can we also add a sentence about the case when it returns an empty bytearray? Something like:

Returns an empty bytearray if any of the images cannot be retrieved.

This brings another question (which also can be addressed later) - should it return an empty value or would raising an exception be a better choice? Returning empty doesn't really tell the caller that something failed so that error might go unnoticed and the program will continue executing failing in an unexpected way without a good error message.

Reply:
Raising an Exception might be better here.  I think it makes the logic a bit easier to handle in the event one of the images is unavailable.

I'll update that and the docs.